aten/src/ATen/native/cuda/BinaryMiscOpsKernels.cu

```diff
@@ -137,9 +137,9 @@ void logical_xor_kernel_cuda(TensorIterator& iter) {
 }
 
 void smooth_l1_kernel_cuda(TensorIterator& iter) {
-  AT_DISPATCH_ALL_TYPES_AND(kHalf, iter.dtype(), "smooth_l1_cuda", [&]() {
-    gpu_kernel(iter, [] GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
-      auto z = fabs(a - b);
```

Comment:
`fabs` is for `double`, I didn't see any need to use `double` instead of `scalar_t`